cv_next/components/layout/navbar/popup.tsx

```diff
@@ -78,18 +79,23 @@ const UserDataComponent: React.FC<{
 export default function Popup({ closeCb, userData, updateSignIn }: PopupProps) {
   const dialogRef = useRef<HTMLDialogElement>(null);
   const supabase = createClientComponent();
+  const fetchFromApi = useApiFetch();
 
   useEffect(() => {
     if (dialogRef.current) {
       dialogRef.current.show();
     }
   }, []);
 
-  const handleSelection = (route: string) => {
+  const handleSelection = async (route: string) => {
     if (route === "Signout") {
-      supabase.auth.signOut();
+      await supabase.auth.signOut();
+      await fetchFromApi(
```

Comment:
Actually I think this is handled in the new `useUser` hook, before merging this please check the behavior in #99

Reply:
The signout must happen using a supabase client component to delete the cookies automatically
Do you mean the API call/updateSignIn? because yeah maybe we could signout and manually revalidate

Reply:
I meant that after signing-out and re-rendering homepage, the `useUser` hook mutates and calls the revalidation itself. That's at least how I remember it worked for me locally, so please checkout the #99 branch and see if that's the behavior you expect.

Reply:
Ah I now see, in the #99 branch I added a call to `mutateUser` from the `useUser` hook. I believe it's better that way because we keep the SWR cache synced with our actions.
In general - I think every user-related action should be accessed/operated through the `useUser` now, so that the SWR will be up-to-date.
In merge conflicts in #99 I'll keep the `mutateUser` call, after merging it feel free to adjust if needed.

Reply:
That's odd, currently in prod without the revalidate we must refresh after the signout
Maybe because the forced revalidate is missing or something?(like the home page isn't re-rendered)

Reply:
Prod includes #99?

Reply:
Yes, was probably merged when the dev pipeline was created(when we synced the branches)

Reply:
My mistake: in prod it does remove your data but the popup doesn't change until refresh(or a manual revalidation is needed)

Reply:
Hmm.. Weird. I'll check it tomorrow, bc that PR included force revalidation on the signout route

Reply:
OK I see the issue, it's an issue of the #99 code not of the signout page. I think it'd be better to open a PR for that instead (I'll do it). The issue is that I've introduced retrying which meant apparently that on error we keep the data we currently have and never replace it with empty data.

Reply:
Please check #114

Reply:
I see
Well, the changes in #155 good seem and did fix the issue with the manual mutation on signout which was what I aimed for here so should I merge that and close this one?

Reply:
Yeah that's what I meant. Thanks